examples/route-list.rs

```diff
@@ -0,0 +1,125 @@
+extern crate buffering;
+extern crate neli;
+
+use buffering::StreamReadBuffer;
+use neli::consts::*;
+use neli::nl::NlEmpty;
+use neli::nl::Nlmsghdr;
+use neli::rtnl::*;
+use neli::socket::*;
+use neli::Nl;
+use neli::MAX_NL_LENGTH;
+use std::net::IpAddr;
+
+
+///
+/// This sample is a simple imitation of the `ip route` command, to demonstrate interaction with the rtnetlink subsystem.  
+/// 
+fn main() {
+    let mut socket = NlSocket::connect(NlFamily::Route, None, None, true).unwrap();
+
+    let rtmsg : Rtmsg<Rta> = Rtmsg {
+        rtm_family: libc::AF_INET as u8,
+        rtm_dst_len: 0,
+        rtm_src_len: 0,
+        rtm_tos: 0,
+        rtm_table: RtTable::Unspec,
+        rtm_protocol: Rtprot::Unspec,
+        rtm_scope: RtScope::Universe,
+        rtm_type: Rtn::Unspec,
+        rtm_flags: vec![],
+        rtattrs: vec![],
+    };
+    let nlhdr = {
+        let len = None;
+        let nl_type = Rtm::Getroute;
+        let flags = vec![NlmF::Request, NlmF::Root, NlmF::Match];
+        let seq = None;
+        let pid = None;
+        let payload = rtmsg;
+        Nlmsghdr::new(len, nl_type, flags, seq, pid, payload)
+    };
+    socket.send_nl(nlhdr).unwrap();
+
+    'socket_read: loop {
+        let mut mem = vec![0u8; MAX_NL_LENGTH];
+        match socket.recv(&mut mem, 0) {
```

Comment:
So I'm reviewing now. Was there a reason that you used `recv` instead of `recv_nl` here? Much of the following lines look like they are a reimplementation of `recv_nl`.

Reply:
It was to allow a reinterpretation of the received buffer, rather than discarding it, when looking at a message.

Reply:
Okay, that makes sense. I would recommend doing it the way I have done it until #19 is complete. That should ease some of this.